lib/omise/OmiseSearch.php

```diff
@@ -0,0 +1,224 @@
+     *
+     * @param  string $filters  Searching text with specific key within the scope.
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function filter(array $filters = array())
+    {
+        foreach ($filters as $k => $v) {
+            if (is_bool($v)) {
+                $filters[$k] = $v ? 'true' : 'false';
+            }
+        }
+        return $this->mergeAttributes('filters', $filters);
+    }
+
+    /**
+     * Update `page` parameter.
+     *
+     * @param  string $page  Specific number of searching page.
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function page($page)
+    {
+        return $this->mergeAttributes('page', $page);
+    }
+
+    /**
+     * Update `order` parameter.
+     *
+     * @param  string $order  The order of the list returned.
+     *
+     * @see https://www.omise.co/search-api
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function order($order)
+    {
+        return $this->mergeAttributes('order', $order);
+    }
+
+    /**
+     * Check whether this search instance is dirty or not.
+     *
+     * Dirty search instance needs to be retrieved before further usage. The
+     * dirty search instance automatically retrieves its data when client code
+     * tries to access its array element.
+     *
+     *     $search = OmiseCharge::query('demo'); // the instance is dirty
+     *     echo $search['object'];               // this will automatically retrive remote value
+     *
+     * @return  bool  true if the instance is dirty and needs to be retrieved
+     */
+    public function isDirty() {
+        return $this->dirty;
+    }
+
+    /**
+     * Retrieve search data from Omise server.
+     *
+     * This method does nothing if the current instance is not dirty.
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function retrieve()
+    {
+        if (!$this->dirty) {
+            return;
+        }
+
+        $this->g_reload($this->getUrl());
+        $this->dirty = false;
+        return $this;
+    }
+
+    /**
+     * Merge the given key and value to search attributes, and set instance state
+     * as dirty.
+     *
+     * @param  string  $key  Search attribute key.
+     * @param  mixed  $value  Search attribute value.
+     */
+    private function mergeAttributes($key, $value)
+    {
+        $this->dirty = true;
+        $this->attributes[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Generate request url.
+     *
+     * @param  string $querystring
+     *
+     * @return string
+     */
+    private function getUrl()
+    {
+        $querystring = http_build_query($this->attributes);
+        return OMISE_API_URL.self::ENDPOINT.'/?'.$querystring;
+    }
+
+    // Override methods of ArrayAccess
+
+    /*
+     * (non-PHPdoc)
+     *
+     * @see  OmiseObject::offsetSet()
+     */
+    public function offsetSet($key, $value)
+    {
+        $this->retrieve();
```

Comment:
@iwat Can you also provide `reload()` method? to make the usage the same as other classes.
And maybe these class can also execute `$this->reload()` instead of `$this->retrieve()`?